### PR TITLE
[FOGL-7247]: Codemirror editor width issue for large content

### PR DESCRIPTION
**Pull request by monika-Dianomic**

### CODE REVIEW

File: src/app/components/core/configuration-manager/view-config-item/view-config-item.component.css [hunk 1]
```diff
@@ -87,6 +87,10 @@
   cursor: pointer !important;
 }
 
+.codemirror-field {
+  width: 610px;
```
Comment (praveen-garg) [inline src/app/components/core/configuration-manager/view-config-item/view-config-item.component.css:91]: Tested on different browsers?

File: src/app/components/core/configuration-manager/view-config-item/view-config-item.component.html [hunk 2]
```diff
@@ -27,7 +27,7 @@
                           [disabled]="configItem?.value?.hasOwnProperty('editable') && !configItem?.value?.editable" />
                       </label>
                     </ng-container>
-                    <div class="field">
+                    <div class="field" [ngClass]="{'codemirror-field': getConfigAttributeType(configItem.value.type) === 'LONG_TEXT' || getConfigAttributeType(configItem.value.type) === 'RAW_TEXT' || getConfigAttributeType(configItem.value.type) === 'SCRIPT'}">
```
Comment (praveen-garg) [inline src/app/components/core/configuration-manager/view-config-item/view-config-item.component.html:30]: Better use includes?

Comment (praveen-garg) [reply src/app/components/core/configuration-manager/view-config-item/view-config-item.component.html:30]: ('LONG_TEXT', 'RAW_TEXT', 'SCRIPT').includes(getConfigAttributeType(configItem.value.type))

JSON ?

File: src/app/components/core/configuration-manager/view-config-item/view-config-item.component.css (hunk 1 above)
Comment (monika-Dianomic) [reply src/app/components/core/configuration-manager/view-config-item/view-config-item.component.css:91]: Yes, I've tested it on Safari and Chrome browser. It is working fine on both.

File: src/app/components/core/configuration-manager/view-config-item/view-config-item.component.ts [hunk 3]
```diff
@@ -149,6 +149,12 @@ export class ViewConfigItemComponent implements OnInit, OnChanges, OnDestroy {
       this.checkValidityOnPageLoad();
     }
     this.cdRef.detectChanges();
+
+    // get width of codemirror textarea (if exist)
+    let codemirrorTextarea = document.getElementById('cdmr-field');
```
Comment (praveen-garg) [inline src/app/components/core/configuration-manager/view-config-item/view-config-item.component.ts:154]: with ID there can be multiple fields in DOM.

File: src/app/components/core/configuration-manager/view-config-item/view-config-item.component.html [hunk 4]
```diff
@@ -27,7 +27,7 @@
                           [disabled]="configItem?.value?.hasOwnProperty('editable') && !configItem?.value?.editable" />
                       </label>
                     </ng-container>
-                    <div class="field">
+                    <div class="field" [style.max-width.px]=cdmrTextareaWidth id="cdmr-field" [ngClass]="{'codemirror-field': getConfigAttributeType(configItem.value.type) === 'LONG_TEXT' || getConfigAttributeType(configItem.value.type) === 'RAW_TEXT' || getConfigAttributeType(configItem.value.type) === 'SCRIPT'}">
```
Comment (praveen-garg) [inline src/app/components/core/configuration-manager/view-config-item/view-config-item.component.html:30]: avoid ID? e.g. in python35 filter (and/or + South plugins with such fields) will cause multiple objects with same ID in DOM.

**Review by mshariq-nerd — APPROVED** [review]

**Review by praveen-garg — APPROVED** [review]

**Pull request merged**